moto/dynamodb2/models.py

```diff
@@ -110,27 +111,27 @@ def describe_attrs(self, attributes):
         }
 
     def update(self, update_expression, expression_attribute_names, expression_attribute_values):
-        ACTION_VALUES = ['SET', 'set', 'REMOVE', 'remove']
-
-        action = None
-        for value in update_expression.split():
-            if value in ACTION_VALUES:
-                # An action
-                action = value
-                continue
-            else:
+        parts = [p for p in re.split(r'\b(SET|REMOVE|ADD|DELETE)\b', update_expression) if p]
```

Comment:
`'SET #c0=:v0, #c1=:v1, #c...2, #c3=:v3, #c4=:v4, #c5=:v5, #c6=:v6, #c7=:v7, #c8=:v8, #c9=:v9, #c10=:v10'`, for example, generates the list `[u'', u'SET', u' #c0=:v0, #c1=:v1, #c2=:v2, #c3=:v3, #c4=:v4, #c5=:v5, #c6=:v6, #c7=:v7, #c8=:v8, #c9=:v9, #c10=:v10']`, so this `if p` filters out the empty starting element. I guess it's true that only that starting element should ever be empty - I could change the code to check that explicitly and then throw it out?